ui/scripts/system.js

```diff
@@ -14088,6 +14093,12 @@
                             }
                         });
                     },
+                    actionPreFilter: function(args){
+                        if (g_enablemetricsui)
+                            return ['add', 'viewMetrics'];
+                        else
+                            return ['add'];
+                    },
```

Comment:
Lines 14096-14101 appear to be a duplication of lines 7857-7860.  Can this code duplication be removed?